LiskHQ/lisk
# Apply lisk eslint config - Closes #1207

### What was the problem?
Lisk core currently using basic eslint config which does not include most of the standard eslint rules, as part of this pull request we are going to extend `eslint-config-lisk-base` and implement the eslint rules.
### How did I fix it?
Included `.eslintrc.json` at root and `.eslintrc.json` in the test directory for test specific rules.
Introducing `lint` and `lint:fix` npm run scripts and `lint` runs against git `precommit` and `prepush`
### How to test it?
`npm run lint` to see all the lint issues.
`npm run lint:fix` will automatically fix certain issues.
### Review checklist

* The PR solves #1207 
* All new code is covered with unit tests
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated
